mlflow2seldon/api.py

```diff
@@ -0,0 +1,291 @@
+                            model_storage_uri=storage_art_uri,
+                            model_name=model.name,
+                            model_stage=model_version.current_stage,
+                            model_version=model_version.version,
+                            source_run_id=model_version.run_id,
+                            deployment_namespace=seldon_deployment_ns,
+                        )
+                        deployed_model = seldon_models.get(registered_model.name)
+                        if not deployed_model or not deployed_model.is_same_version(
+                            registered_model
+                        ):
+                            registered_model.need_redeploy = True
+                        else:
+                            registered_model.need_redeploy = False
+                        mlflow_models[registered_model.name] = registered_model
+
+                # deploy models in Seldon
+                for model_name in mlflow_models.keys():
+                    if mlflow_models[model_name].need_redeploy:
+                        await _deploy_model(
+                            mlflow_models[model_name],
+                            neuro_client,
+                            registry_secret_name,
+                        )
+                        mlflow_models[model_name].need_redeploy = False
+
+                # removing outdated models
+                for model_name in set(seldon_models.keys()) - set(mlflow_models.keys()):
+                    _delete_seldon_deployment(seldon_models[model_name])
+
+                # the state is sync
+                seldon_models = mlflow_models.copy()
+                deployed_models = [
+                    f"{n}:{m.model_version}" for n, m in seldon_models.items()
+                ]
+                logging.info(f"Deployed models: {';'.join(deployed_models)}")
+                await asyncio.sleep(DELAY)  # not to overload an API
+
+            except KeyboardInterrupt:
+                logging.warning("Got keyboard interrupt, gracefully shutting down...")
+                break
+            except MlflowException as e:
+                if "Page Not Found" in e.message:
+                    logging.warning(
+                        "Unable to connect to the MLFlow server. "
+                        "Is it running as a platform job?"
+                    )
+                elif "Sign In" in e.message:
+                    logging.warning(
+                        "Unable to connect to the MLFlow server - auth is required. "
+                        "SSO is disabled?"
+                    )
+                else:
+                    logging.warning(f"MLFlow server error occured:\n{e}")
+            except Exception as e:
+                logging.warning(f"Unexpected exception (ignoring): {e}")
+    finally:
+        for model in seldon_models.values():
+            _delete_seldon_deployment(model)
+
+
+async def _deploy_model(
```

Comment:
recently I got concerned that we don't test our code anyhow. Since we started "productionize" our demos, we might want to add some tests. So that to create a class `SeldonDeployer` with `_deploy_model()` and _poll_mlflow()` loop. And a unit test inherits this class and overloads the first method so that we don't need K8s to test it. Sth simple as that.

Reply:
to keep track on it, create issue #8